ramiro_serra/Debug and profiling notes.md

```diff
@@ -0,0 +1,113 @@
+# Debug and profiling notes
+The whole document is based on [this note](http://wiki.ros.org/roslaunch/Tutorials/Roslaunch%20Nodes%20in%20Valgrind%20or%20GDB), so you can check it first.
+To be able to follow the examples, install valgrind and callgrind.
+
+## Using Valgrind for profiling
+
+[This other article](http://wiki.ros.org/roslaunch/Tutorials/Profiling%20roslaunch%20nodes) talks about profiling.
+We'll use the [create_autonomy](https://github.com/RoboticaUtnFrba/create_autonomy) repo, in order to make some examples.
+
+First, we open the docker launchfile, [see here](https://github.com/RoboticaUtnFrba/create_autonomy/wiki/Setup-Docker), then we stall `valgrind` and `kcachegrind`, we will make use of that later. `sudo apt-get install valgrind callgrind -y`
+
+Then, we compile our project, `catkin_make DCMAKE_BUILD_TYPE=Debug`.
+Then, we source devel/setup.bash `source devel/setup.bash`
+Now, we will decide which node to profile, in our case `send_goal`, a node located in `ca_move_base` pkg.
+In order to profile that node, we have to include a `launch-prefix` tag, into the desired `node` tag into the `launch file`.
+So, to locate the launchfile `roscd ca_move_base/launch`.
+We can edit the `send_goal.launch` file with whatever editor we feel like.
+
+This is how my `send_goal.launch` looked like after the changes I made.
+    
+    <launch>
+        <arg name="x" default="2.0"/>
+        <arg name="y" default="2.0"/>
+        <arg name="Y" default="1.0"/>
+        <arg name="ns" default="create1"/>
+        <node pkg="ca_move_base" type="send_goal" name="navigation_goals"
+                args="$(arg x) $(arg y) $(arg Y)" ns="$(arg ns)" launch-prefix="valgrind --tool=callgrind --callgrind-out-file='callgrind.sendgoal.%p'"/>

```

Comment:
What about this?
```xml
<arg name="debug" default="false"/>
<arg unless="$(arg debug)" name="launch_prefix" value=""/>
<arg          if="$(arg debug)" name="launch_prefix" value="valgrind --tool=callgrind --callgrind-out-file='callgrind.sendgoal.%p'"/>

<node ... launch-prefix="$(arg launch_prefix)"/>
```

It seems a much cleaner way of doing it and it's based on `gazebo_ros`' launch file.

Reply:
Adressed